lib/practice_exercises.rb

```diff
@@ -1,13 +1,40 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  alph_str = string.downcase.gsub(/[^a-z\w]/,'') #substitute any non-letter or whitespace with nothing
+  alph_str.length/2.times do |index|
+    if alph_str[index] != alph_str[-1 - index]
+      return false
+    end
+    return true
+  end
+
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
This works, but your time complexity is going to be O(n * m) where n is the number of strings and m is the length of the shortest string.